src/Sylius/Component/Core/Model/ProductVariant.php

```diff
@@ -173,7 +173,10 @@ public function getPrice()
      */
     public function setPrice($price)
     {
-        $this->price = (int)$price;
+        if (!is_int($price)) {
+            throw new \InvalidArgumentException('Price must be an integer.');
```

Comment:
IMHO this is a crucial thing to not allow for setting other value than int. It could break the whole money logic and cause disastrous consequences. I will leave it for now, until we decide to use a ValueObject for money management and have a type hint in the function parameter.